rfc/2023/2023-05-01-va-rfc-lightweight-apps-paths.md

```diff
@@ -0,0 +1,91 @@
+
+
+## Design
+
+> Explain the proposed design in enough detail so that a team member will fully understand the implementation. Include a diagram (in the `images` dir) as needed to convey your plans. Use active voice, present tense, and decisive language.
+
+The Platform will provide a minimalist option for application teams to create new applications that prioritize generating HTML at build time and do not require React. To start, this can be done with small adjustments to existing tools, e.g. webpack, to allow application manifests to specify non-JSX entry points, exclude React and related libraries, and allow for VFS teams to use/extend existing EJS templates.
+
+VFS teams using the minimalist option will need to adopt a [performance budget](va-rfc-frontend-performance.md), and ensure that complex logic and data manipulation is performed server-side. Minimalist frontends should be able to fetch data that is immediately renderable to the end user. Since `vets-api` already is already responsible for managing and processing dynamic and personalized data, as well as feature toggles, the backend is already well-suited for generating ready-to-display data for frontends.
+
+Representatives from VFS teams and the Platform should work together to explore what resources might aid a minimalist application; given that the design system is using web components, are there tools or guidance we might want to adopt for when and how to use web components?
+
+VFS and Platform engineers should also look into what tools might help existing applications achieve better performance by shifting more logic to `vets-api` and HTML generation to build time. Frameworks like Next.js support [static HTML export](https://nextjs.org/docs/advanced-features/static-html-export), which might provide a migration path for existing applications.
+
+## Risks
+
+
+>List the risks of this approach
+> * There are always risks. What are the risks of this solution?
+> * These are the things people will bring up in opposition to your idea or plans. Acknowledge them.
+
+
+- There aren't enough guide rails without a framework like React.
+    - Platform and VFS teams should evaluate minimalist JS frameworks ( [alpinejs](https://alpinejs.dev), [htmx](https://htmx.org), etc.) for lightweight interactivity/dynamic content.
+    - Platform and VFS teams can collaborate on best practices and update the Platform documentation to reflect those best practices. Those best practice could also be incorporated into issue templates and collab cycle reviews.
+- SSG will require lots of additional tools, creating its own complexity
+    - `vets-website` already has support for ejs templates, and app entry points don't intrinsically have to use React, so we already have the tools we need
+    - Next.js provides SSG tools for React applications, and that's something that could be explored further
```

Comment:
It'd be interesting to see if we could use Next.js 13+ with Module Federation to import the different applications.